zerver/tests/test_users.py

```diff
@@ -3195,6 +3195,32 @@ def test_get_inaccessible_user_ids(self) -> None:
         )
         self.assertEqual(inaccessible_user_ids, {othello.id})
 
+    def test_get_users_for_spectators(self) -> None:
+        # Checks that spectators can fetch users data.
+        hamlet = self.example_user("hamlet")
+        othello = self.example_user("othello")
+
+        with self.assert_database_query_count(4):
+            result = orjson.loads(self.client_get("/json/users").content)
```

Comment:
We should always `assert_json_success(result)` before trying to load JSON on it.